[Eugene Tay] IP

eugenetayyj opened the pull request:
## BOTCHAT123

> “Ideas without action are worthless.” – Harvey Mackay ([source](https://www.quotemaster.org/qd5032eddfeb57cd161722ebf874ebfd1))

Botchat123 frees your mind of having to remember things you need to do. It is

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/eugenetayyj/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is FREE!

Features:
- [x]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you're a Java programmer, you can use it to practice Java too. Here's the main method:

`public static void main(String[] args) {
        Duke duke = new Duke("./data/Duke.txt");
    }`

Get your BotChat123 here **today**!

Commit:
Add Gradle support

eugenetayyj committed:
level- 1

eugenetayyj committed:
Level-2

eugenetayyj committed:
Level -3

eugenetayyj committed:
Level-4

eugenetayyj committed:
A-TextUiTesting

eugenetayyj committed:
Level-5

eugenetayyj committed:
Level-6

eugenetayyj committed:
Added javadocs

eugenetayyj committed:
Level-6

eugenetayyj committed:
Level-7

eugenetayyj committed:
Level-7

eugenetayyj committed:
Level-7

eugenetayyj committed:
Level-7

eugenetayyj committed:
Level-8

eugenetayyj committed:
Merge branch 'branch-Level-7'

eugenetayyj committed:
Level-7

eugenetayyj committed:
Level-8

eugenetayyj committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

eugenetayyj committed:
A-Gradle

eugenetayyj committed:
A-MoreOOP

eugenetayyj committed:
A-Packages

eugenetayyj committed:
A-JUnit

eugenetayyj committed:
A-JavaDoc

eugenetayyj committed:
Level-9

eugenetayyj committed:
Level-9

eugenetayyj committed:
Merge branch 'branch-A-JavaDoc'

eugenetayyj committed:
Level-9

eugenetayyj committed:
A-CheckStyle

shwene commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,51 @@
+                }
+                Command command = parser.parse(inputCommand);
+                command.execute(taskList, ui, storage);
+            } catch (InputMismatchException | IndexOutOfBoundsException
+                    | NumberFormatException | NullPointerException e) {
```
Was bitwise or `|` intended here instead of logical or `||` ?

shwene commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,51 @@
+                command.execute(taskList, ui, storage);
+            } catch (InputMismatchException | IndexOutOfBoundsException
+                    | NumberFormatException | NullPointerException e) {
+                new ErrorCommand().execute(taskList, ui, storage);
```
Perhaps the specific error could be displayed to the user

shwene commented on src/main/java/duke/main/Parser.java:
```diff
@@ -0,0 +1,62 @@
+
+/**
+ * Represents a Parser that determines what kind of input was obtained
+ * from the user and then decisively returns different commands
```
Perhaps singular form of command is meant here?

shwene commented on src/main/java/duke/main/Parser.java:
```diff
@@ -0,0 +1,62 @@
+                } else if (command.equals("event")) {
+                    return new AddEventsCommand(task, inputs[3]);
+                } else if (command.equals("delete")) {
+                    return new DeleteCommand(inputs[1]);
```
Perhaps these "magic numbers" can be defined as static constants at the top, e.g. INDEX = 1

shwene commented on src/main/java/duke/main/Ui.java:
```diff
@@ -0,0 +1,64 @@
+     *
+     * @param task
+     */
+    public void repeater(String task) {
```
Perhaps `echoTask(...` would be more descriptive

shwene reviewed (commented):
Good job! Just a few minor nits.

eugenetayyj committed:
Level-10

eugenetayyj committed:
Level-10

eugenetayyj committed:
Level-10

eugenetayyj committed:
Level-10

eugenetayyj committed:
Generate JAR

eugenetayyj committed:
A-Assertions

eugenetayyj committed:
A-CodeQuality

eugenetayyj committed:
Added assertions

eugenetayyj committed:
Merge branch 'master' into branch-A-Assertions

eugenetayyj committed:
Merge pull request #1 from eugenetayyj/branch-A-Assertions

A-Assertions

eugenetayyj committed:
Merge pull request #2 from eugenetayyj/branch-A-CodeQuality

A-CodeQuality

eugenetayyj committed:
Assertions added to crucial places like in loading tasks in Storage, event and deadline constructor. These areas are more susceptible to errors if there are changes to the system.

eugenetayyj committed:
Improved code quality by avoiding long methods and avoiding complicated expressions. This makes the code more readable

eugenetayyj committed:
A-CI

eugenetayyj committed:
C-Archive

Implemented an archive system with another instance of storage and tasklist to save. The deleted data is stored in an archive text file.

eugenetayyj committed:
Merge pull request #3 from eugenetayyj/branch-C-Archive

C-Archive

eugenetayyj committed:
Update DialogBox.java

Improved code quality by removing presence of magic numbers of how the image in the bot is centre and its radius.
Magic number is replaced with a constant and the constant is used instead.

eugenetayyj committed:
Change Font

eugenetayyj committed:
Add Title Icon

eugenetayyj committed:
Fix Archive Bug

eugenetayyj committed:
Add UG

eugenetayyj committed:
Reformat UG

eugenetayyj committed:
Fix Git Issues

eugenetayyj committed:
Fix UG Table

eugenetayyj committed:
Change JavafX Version

eugenetayyj committed:
Change UI
